packages/engine/errors/2414.md

```diff
@@ -0,0 +1,12 @@
+---
+original: "Class name cannot be '{0}'"
+excerpt: "You can't give a class the name of '{0}' because it's protected by TypeScript."
+---
+
+Instead of naming your classes based on TypeScript types,
+
+```ts
+class {0} {
+}
+
+You should name your classes to something which is not a typescript type. Checkout types here, [Everyday Types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html)
```

Comment:
```suggestion
Instead of naming your classes based on TypeScript types...

```ts
class {0} {
}

...you should name your classes to something which is not a TypeScript type. Check out all the types in the docs on [Everyday Types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html).
```